### PR TITLE
Fix missing `SegmentDescription` in `PydicomReader`

### LINKED ISSUE #6928: Get an error when reading dicom segmentation
**Describe the bug**
When I tried to read the dicom segmentation label with LoadImage in MONAI transforms, it reported a bug shown below.

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/venv/lib/python3.11/site-packages/monai/transforms/io/array.py", line 290, in __call__
    img_array, meta_data = reader.get_data(img)
                           ^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/monai/data/image_reader.py", line 586, in get_data
    data_array, metadata = self._get_seg_data(d)
                           ^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/monai/data/image_reader.py", line 770, in _get_seg_data
    class_name = description.SegmentDescription
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/pydicom/dataset.py", line 908, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Dataset' object has no attribute 'SegmentDescription'
```

**To Reproduce**
1. Install the MONAI main branch.
2. Download the attachment [dcm file](https://drive.google.com/file/d/13_-1weQkBlkbcMzWixqKUDCGBs1lu2oq/view?usp=sharing) whose md5 is f511acb750076c56ad2d9647dcfe521d.
3. Run command `python -c "from monai.transforms import LoadImage; loader = LoadImage(reader='pydicomreader'); loader('./SE000000.dcm');"`
4.  If I changed [this line](https://github.com/Project-MONAI/MONAI/blob/e48c3e2c741fa3fc705c4425d17ac4a5afac6c47/monai/data/image_reader.py#L769) to `class_name = description.SegmentDescription if hasattr(description, "SegmentDescription") else description.SegmentLabel`, it works. However I am not sure if this modification is good enough.

**Expected behavior**
Read segmentation correctly.

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #6928.

### Description

`SegmentDescription` is optional, https://dicom.innolitics.com/ciods/segmentation/segmentation-image/00620002/00620006, so try [`SegmentLabel`](https://dicom.innolitics.com/ciods/segmentation/segmentation-image/00620002/00620005) first.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -766,7 +766,9 @@ def _get_seg_data(self, img):
             all_segs = np.zeros([*spatial_shape, n_classes])
 
         for i, (frames, description) in enumerate(self._get_frame_data(img)):
-            class_name = description.SegmentDescription
+            class_name = (
+                description.SegmentLabel if hasattr(description, "SegmentLabel") else description.SegmentDescription
+            )
```
Comment (wyli) [inline monai/data/image_reader.py:771]: Still has the potential issue of AttributeError: 'Dataset' object has no attribute 'SegmentDescription'?

**Review by wyli — APPROVED** [review]

File: monai/data/image_reader.py [hunk 2]
```diff
@@ -766,7 +766,11 @@ def _get_seg_data(self, img):
             all_segs = np.zeros([*spatial_shape, n_classes])
 
         for i, (frames, description) in enumerate(self._get_frame_data(img)):
-            class_name = description.SegmentDescription
+            class_name = (
+                description.SegmentDescription
```
Comment (ericspod) [inline monai/data/image_reader.py:770]: Could this be done as 'getattr(description, "SegmentDescription", description.SegmentLabel)`? This also assumes the `SegmentLabel` property is always present.

**Review by ericspod — APPROVED** [review]
Looks good to me!

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**